### PR TITLE
fix: culling controller mesh update issue

### PR DESCRIPTION
**Pull request by BrianAmadori**
addresses #1741

The fix just ensures `MarkDirty` is called when it should.

`MarkDirty` was missing on:
- Entity movement
- Pool `Get`

Also, dirtyness is now decoupled. Now has independent values for refilling the renderer list and updating the culling for the existing list.

### CODE REVIEW

File: unity-client/Assets/Rendering/Culling/CullingController.cs [hunk 1]
```diff
@@ -260,6 +260,8 @@ internal void SetCullingForRenderer(Renderer r, bool shouldBeVisible, bool shoul
             {
                 if (r.forceRenderingOff != !shouldBeVisible)
                     r.forceRenderingOff = !shouldBeVisible;
+
+                Debug.Log($"-- Asd: {r.name} ... {r.forceRenderingOff}");
```
Comment (AjimenezDCL) [inline unity-client/Assets/Rendering/Culling/CullingController.cs]: Remove debug

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/ParcelScene.cs [hunk 2]
```diff
@@ -488,50 +485,49 @@ public void SetEntityParent(string entityId, string parentId)
...
-                    // In this case, the entity will attached to the first person camera
-                    // On first person mode, the entity will rotate with the camera. On third person mode, the entity will rotate with the avatar
-                    me.SetParent(DCLCharacterController.i.firstPersonCameraReference);
-                    SceneController.i.boundariesChecker.AddPersistent(me);
-                    Environment.i.physicsSyncController.MarkDirty();
+                    SceneController.i.boundariesChecker.RemoveEntityToBeChecked(me);
                 }
-                else if (parentId == "AvatarEntityReference" || parentId == "AvatarPositionEntityReference") // AvatarPositionEntityReference is for compatibility purposes
+
+                if (parentId == "0")
```
Comment (AjimenezDCL) [inline unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/ParcelScene.cs:512]: Just curious about this comparison to "0". Mind to elaborate?

**Review by AjimenezDCL — APPROVED** [review]

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/ParcelScene.cs (hunk 2 above)
Comment (BrianAmadori) [reply unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/ParcelScene.cs:512]: When a `SetEntityParent` message comes with `0` as payload, this means that the entity should be attached to the "root" of the scene. Like setting `parent == null` in Unity.

File: unity-client/Assets/Rendering/Culling/CullingController.cs (hunk 1 above)
Comment (BrianAmadori) [reply unity-client/Assets/Rendering/Culling/CullingController.cs]: Done.

**Review by pravusjif — APPROVED** [review]
the build was "buttery smooth"

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/ParcelScene.cs (hunk 2 above)
Comment (AjimenezDCL) [reply unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/ParcelScene.cs:512]: Maybe we can have a comment or have a static method `ParentIdIsRoot` that does the comparison.

**Pull request merged**